src/MiniCover.HitServices/HitTestMethod.cs

```diff
@@ -76,7 +76,9 @@ public static IEnumerable<HitTestMethod> MergeDuplicates(IEnumerable<HitTestMeth
 
         public void Serialize(Stream stream)
         {
+            Dictionary<int, int> hits;
```

Comment:
Sorry, I prefer the way it was.

Now I get why you have that lock. It's to avoid issues with iterator. But when this method is called the dictionary shouldn't be modified anymore. If it is being modified I think we have an issue somewhere else.

I think it's important to protect against 2 HitTestMethod writing to same file concurrently.

What do you think about?
```
lock(_lock)
lock(stream)
using (
```

Reply:
We don't have 2 file so no need  to lock the stream
Where I saw this problem is a method that is doing a Fire And Forget operation and then this operation is hit during the serialize. We can't do anything for that but writing in a memory stream and then in the file, if we we have a hit during serialization we start it again.

Reply:
no need to lock the stream as it is already locked in the calling method so I revert back to previous status

Reply:
Yes that could be something but how you start it and how you stop it.
One way would be to have only one command which take the tests execution command as parameter. I m currious to see how any coverage tool is dealing with the fire and forget

Reply:
I was thinking about a completely separate command:
```
minicover server start
..run tests a
..run tests  b
..run application
minicover server stop
```